### PR TITLE
query for allow all IAM policies

### LINKED ISSUE #1378: Add Allow all IAM policies query for Ansible
### Platform
Ansible

### Provider
AWS

### Description
IAM policies allow all ('*') in a statement action

### PR DESCRIPTION
**Pull request by soniantao**
Closes #1378
Closes #1379

### CODE REVIEW

File: assets/queries/ansible/aws/IAM_policy_all/query.rego [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package Cx
+
+CxPolicy [  result ] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  awsApiGateway := task["community.aws.iam_managed_policy"]
+  contains(awsApiGateway.state, "present")
+  resource := awsApiGateway.policy.Statement[_].Resource
+  contains(resource, "*")
```
Comment (joaomartinscx) [inline assets/queries/ansible/aws/IAM_policy_all/query.rego:10]: Besides the policy affecting all Resources, the query checks if the effect is to allow some action in them. You need to check if the Effect is Allow as well

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/ansible/aws/IAM_policy_all/metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "Allow_all_IAM_policies",
+  "queryName": "Allow all IAM policies",
```
[the remark this answers is not in the document]
Comment (soniantao) [reply assets/queries/ansible/aws/IAM_policy_all/metadata.json:3]: You are correct. I've just follow Rui's instructions to keep the name & id. However, after talk with him, I'll change both queries: ansible and terraform to keep the same structure

File: assets/queries/ansible/aws/IAM_policy_all/query.rego [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package Cx
+
+CxPolicy [  result ] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  awsApiGateway := task["community.aws.iam_managed_policy"]
+  contains(awsApiGateway.state, "present")
```
[the remark this answers is not in the document]
Comment (soniantao) [reply assets/queries/ansible/aws/IAM_policy_all/query.rego:8]: Once again, you are right. However in terraform query that validation was not included. I'll change both

File: assets/queries/ansible/aws/IAM_policy_all/query.rego (hunk 1 above)
Comment (soniantao) [reply assets/queries/ansible/aws/IAM_policy_all/query.rego:10]: Once again, you are right. However in terraform query that validation was not included. I'll change both

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

**Review by joaomartinscx — APPROVED** [review]
Everything looks okay, good job!

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049036&projectid=718)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/Feature/IAM_all/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049036&projectid=718&pathid=1)
[83](https://github.com/Checkmarx/kics/blob/Feature/IAM_all/test/main_test.go#L83) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049036&projectid=718&pathid=3)
[52](https://github.com/Checkmarx/kics/blob/Feature/IAM_all/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049036&projectid=718&pathid=2)

**Pull request merged**